lib/AST/ASTMangler.cpp

```diff
@@ -635,6 +645,12 @@ void ASTMangler::appendDeclName(const ValueDecl *decl) {
     return appendOperator("L", Index(decl->getLocalDiscriminator()));
   }
 
+  if (customNameAttr && customNameAttr->RelatedEntityKind) {
+    return appendOperatorParam("L",
+                               StringRef(&customNameAttr->RelatedEntityKind,
+                                         1));
```

Comment:
Seems to me that, even in your representation, there should be an accessor on the attribute that returns a StringRef.

Reply:
Fair enough.